doc/client_design_doc.md

```diff
@@ -0,0 +1,56 @@
+# SQLFlow Client Design Doc
+
+## Overview
+
+SQLFlow Client implements a minimum client that issues queries to and fetch results/logs from [sqlflowserver](https://github.com/wangkuiyi/sqlflowserver).
+
+In `pysqlflow`, the client will be wrapped in the magic command where user directly writes SQL statements.
+
+## Example
+
+```python
+import sqlflow
+
+client = sqlflow.Client(address='localhost:50051')
+
+# Regular SQL
+response = client.query('select * from table')
+for dataframe in response:
+    print(dataframe) # {'y': [4, 5, 6], 'x': [1, 2, 3]}
+
+# ML SQL
+response = client.query('select ... train ...')
```

Comment:
I am thingking adding client.train(...) as wrapper of execute trainning. So does client.predict(...).

Reply:
There is no return value for log as implementation log messages from server using logging